### PR TITLE
Set default active cluster selection policy at Frontend

### PR DESCRIPTION
**Pull request by Shaddoll**
**What changed?**
Set default active cluster selection policy at Frontend instead of History

**Why?**
This will remove the topology couple between region and cluster and simplify the implementation. The active cluster lookup will always be one hop and we don't need restriction on failover topology to prevent loops in lookup.

**How did you test it?**
Unit tests and simulation tests

**Potential risks**

**Release notes**

**Documentation Changes**

### CODE REVIEW

File: service/frontend/wrappers/clusterredirection/policy.go [hunk 1]
```diff
@@ -438,44 +438,27 @@ func (policy *selectedOrAllAPIsForwardingRedirectionPolicy) activeClusterForActi
...
-// activeClusterInSameRegion returns the active cluster in the same region as the given cluster name
-// If the lookup fails, it returns current cluster name as fallback
-// In that case, if current cluster is not the active cluster the caller will return an not-active error
-// TODO(active-active): Revisit this logic for strong-consistency requests and potentially reject them instead of serving them from the current cluster
-func (policy *selectedOrAllAPIsForwardingRedirectionPolicy) activeClusterInSameRegion(
-	ctx context.Context,
-	domainEntry *cache.DomainCacheEntry,
-	clusterName string,
-) string {
-	lookupRes, err := policy.activeClusterManager.LookupCluster(ctx, domainEntry.GetInfo().ID, clusterName)
```
Comment (taylanisikdemir) [inline service/frontend/wrappers/clusterredirection/policy.go:475]: you can also remove `LookupCluster` from manager interface/impl

File: service/frontend/wrappers/clusterredirection/api_generated.go [hunk 2]
```diff
@@ -1343,6 +1343,12 @@ func (handler *clusterRedirectionHandler) SignalWithStartWorkflowExecution(ctx c
 
 	var actClSelPolicyForNewWF *types.ActiveClusterSelectionPolicy
 	var workflowExecution *types.WorkflowExecution
+	if sp1.ActiveClusterSelectionPolicy == nil {
+		sp1.ActiveClusterSelectionPolicy = &types.ActiveClusterSelectionPolicy{
+			ActiveClusterSelectionStrategy: types.ActiveClusterSelectionStrategyRegionSticky.Ptr(),
+			StickyRegion:                   handler.GetActiveClusterManager().CurrentRegion(),
+		}
+	}
```
Comment (taylanisikdemir) [inline service/frontend/wrappers/clusterredirection/api_generated.go]: this means we are allowing users to set the `StickyRegion`

**Review by taylanisikdemir — APPROVED** [review]

File: service/frontend/wrappers/clusterredirection/policy.go (hunk 1 above)
Comment (Shaddoll) [reply service/frontend/wrappers/clusterredirection/policy.go:475]: Yeah, I'll do it in a separate PR. I also want to do some refactoring when implementing/fixing lookups for external entity.

File: service/frontend/wrappers/clusterredirection/api_generated.go (hunk 2 above)
Comment (Shaddoll) [reply service/frontend/wrappers/clusterredirection/api_generated.go]: Let me think about it. If we don't want customers to set region in the request, we can check if the policy is sticky region and override the policy set by customers with the current region.

File: common/cache/domainCache.go [hunk 3]
```diff
@@ -1000,6 +1000,7 @@ func (entry *DomainCacheEntry) IsSampledForLongerRetention(
 	return false
 }
 
+// TODO(active-active): This function should accept active cluster selection policy as a parameter
```
Comment (c-warren) [inline common/cache/domainCache.go:1003]: We should figure out if this function makes sense anymore (and in particular if a DomainNotActiveError makes sense) in the active-active world. 

I could see the error being relevant when the cluster is passive in a given failoverGroup. However for standard active-active requests, the workflow could be active in either region depending on the cluster selection policy + secondary attributes. 

(So maybe we also need to have secondary attribute passed here?)

File: service/frontend/wrappers/clusterredirection/policy_test.go [hunk 4]
```diff
@@ -661,11 +651,6 @@ func (s *selectedAPIsForwardingRedirectionPolicySuite) TestActiveClusterForActiv
 				WorkflowID: "wf1",
 			},
 			mockFn: func(activeClusterManager *activecluster.MockManager) {
-				activeClusterManager.EXPECT().LookupCluster(gomock.Any(), domainEntry.GetInfo().ID, s.currentClusterName).Return(&activecluster.LookupResult{
```
Comment (c-warren) [inline service/frontend/wrappers/clusterredirection/policy_test.go:664]: I think we should add a test that passes a nil workflowId to test the new logic

File: common/cache/domainCache.go (hunk 3 above)
Comment (Shaddoll) [reply common/cache/domainCache.go:1003]: Yeah, that's what I put in the comment. 
This function is used in a lot of places. I want to revisit it later.
